yew-ui/src/components/attendants.rs

```diff
@@ -304,57 +321,30 @@ impl Component for AttendantsComponent {
-                                                    .encrypt_with_key(&aes_packet, &pub_key)
-                                                    .map_err(|e| {
-                                                        error!(
-                                                            "Failed to encrypt aes packet: {}",
-                                                            e.to_string()
-                                                        )
-                                                    })
-                                                    .and_then(|data| {
-                                                        ctx.link().send_message(
-                                                            Msg::OnOutboundPacket(PacketWrapper {
-                                                                packet_type: PacketType::AES_KEY
-                                                                    .into(),
-                                                                email: ctx.props().email.clone(),
-                                                                data,
-                                                                ..Default::default()
-                                                            }),
-                                                        );
-                                                        Ok(())
-                                                    });
-                                                Ok(())
-                                            });
-                                            Ok(())
-                                        });
-                                Ok(())
+                        let encrypted_aes_packet = parse_rsa_packet(&response.data)
+                            .and_then(parse_public_key)
+                            .and_then(|pub_key| {
+                                self.serialize_aes_packet()
+                                    .and_then(|aes_packet| Ok((aes_packet, pub_key)))
+                            })
+                            .and_then(|(aes_packet, pub_key)| {
+                                self.encrypt_aes_packet(&aes_packet, &pub_key)
                             });
+
+                        match encrypted_aes_packet {
+                            Ok(data) => {
+                                ctx.link()
+                                    .send_message(Msg::OnOutboundPacket(PacketWrapper {
+                                        packet_type: PacketType::AES_KEY.into(),
+                                        email: ctx.props().email.clone(),
+                                        data,
+                                        ..Default::default()
+                                    }));
+                            }
+                            Err(e) => {
+                                error!("Failed to send AES_KEY to peer: {}", e.to_string());
```

Comment:
This is naive error handling, at least we need to let the user know that they need to refresh the page

Reply:
I created a separate ticket for this

Reply:
I agree!!